client/packages/invoices/src/InboundShipment/DetailView/ContentArea/columns.ts

```diff
@@ -34,230 +39,266 @@ const calculateRowTotalCost = (rowData: InboundLineFragment | InboundItem) => {
-            }
-          },
-          getSortValue: rowData => {
-            if ('lines' in rowData) {
-              const { lines } = rowData;
-              return ArrayUtils.getUnitQuantity(lines);
-            } else {
-              return getUnitQuantity(rowData);
-            }
-          },
-        },
-      ],
+    [
+      'itemUnit',
+      {
+        getSortValue: row =>
+          getColumnPropertyAsString(row, [
+            { path: ['lines', 'item', 'unitName'] },
+            { path: ['item', 'unitName'], default: '' },
+          ]),
+        accessor: ({ rowData }) =>
+          getColumnProperty(rowData, [
+            { path: ['lines', 'item', 'unitName'] },
+            { path: ['item', 'unitName'], default: '' },
+          ]),
+      },
+    ],
+    [
+      'packSize',
+      {
+        accessor: ({ rowData }) =>
+          getColumnProperty(rowData, [
+            { path: ['lines', 'packSize'] },
+            { path: ['packSize'], default: '' },
+          ]),
+        getSortValue: row =>
+          getColumnPropertyAsString(row, [
+            { path: ['lines', 'packSize'] },
+            { path: ['packSize'], default: '' },
+          ]),
+      },
+    ],
+  ];
+
+  if (preferences?.displayVaccineInDoses) {
+    columns.push({
+      key: 'doses',
+      label: 'label.doses-per-unit',
+      sortable: false,
+      accessor: ({ rowData }) => {
+        if ('lines' in rowData) {
+          const { lines } = rowData;
+          if (lines[0]?.item.isVaccine) {
+            const doses = lines?.map(
+              ({ item }) => item?.doses ?? UNDEFINED_STRING_VALUE
+            );
+            const dosesTheSame = doses?.every(dose => dose === doses?.[0]);
+            return dosesTheSame ? doses?.[0] : t('multiple');
+          } else {
+            return UNDEFINED_STRING_VALUE;
+          }
+        } else {
+          return rowData?.item?.isVaccine
+            ? (rowData?.item?.doses ?? UNDEFINED_STRING_VALUE)
+            : UNDEFINED_STRING_VALUE;
+        }
+      },
+    });
+  }
+
+  columns.push(
+    [
+      'numberOfPacks',
       {
-        label: 'label.cost-per-unit',
-        key: 'costPricePerUnit',
-        align: ColumnAlign.Right,
-        width: 120,
-        Cell: CurrencyCell,
         accessor: ({ rowData }) => {
           if ('lines' in rowData) {
             const { lines } = rowData;
-            return ArrayUtils.getAveragePrice(lines, 'costPricePerPack');
+            return ArrayUtils.getSum(lines, 'numberOfPacks');
           } else {
-            return getCostPrice(rowData);
+            return rowData.numberOfPacks;
+          }
+        },
+        getSortValue: rowData => {
+          if ('lines' in rowData) {
+            const { lines } = rowData;
+            return ArrayUtils.getSum(lines, 'numberOfPacks');
+          } else {
+            return rowData.numberOfPacks;
           }
         },
-        sortable: false,
       },
+    ],
+    [
+      'unitQuantity',
```

Comment:
If display in doses is on... wouldn't "dose quantity" be more helpful than unit quantity?

I have about 1300 doses inbound, probably wanting to be able to see that in Detail view? Also, maybe helpful to see dose info in the expand table?

![Screenshot 2025-05-05 at 11 20 56 AM](https://github.com/user-attachments/assets/327caea4-a1f7-42c4-83c5-0a284d134e48)
![Screenshot 2025-05-05 at 11 21 07 AM](https://github.com/user-attachments/assets/be61223b-c67c-46a9-ac52-63cd035c1fd6)